src/js/background.js

```diff
@@ -532,16 +530,22 @@ Badger.prototype = {
    */
   updateBadge: function(tabId){
     if (!this.showCounter()){
-      chrome.browserAction.setBadgeText({tabId: tabId, text: ""});
+      if(chrome.browserAction.setBadgeText){
+        chrome.browserAction.setBadgeText({tabId: tabId, text: ""});
+      }
       return;
     }
     var numBlocked = this.blockedTrackerCount(tabId);
-    if(numBlocked === 0){
-      chrome.browserAction.setBadgeBackgroundColor({tabId: tabId, color: "#00cc00"});
-    } else {
-      chrome.browserAction.setBadgeBackgroundColor({tabId: tabId, color: "#cc0000"});
+    if(chrome.browserAction.setBadgeBackgroundColor){
+      if(numBlocked === 0){
+        chrome.browserAction.setBadgeBackgroundColor({tabId: tabId, color: "#00cc00"});
+      } else {
+        chrome.browserAction.setBadgeBackgroundColor({tabId: tabId, color: "#cc0000"});
+      }
+    }
+    if(chrome.browserAction.setBadgeText){
```

Comment:
Could we move all of these `if (chrome.browserAction....) {` guards to the top of this function?

Also, could we combine these into a global variable like `isMobile`?